Cairo Desktop/Cairo Desktop/DesktopNavigationToolbar.xaml.cs

```diff
@@ -134,53 +133,69 @@ private void LowLevelKeyboardListener_OnKeyUp(object sender, Common.KeyEventArgs
+
+            if (System.Windows.Forms.Screen.AllScreens.Any(s => s.Bounds.Contains((int)(desktopNavigationToolbarLocation.X * DpiHelper.DpiScale), (int)(desktopNavigationToolbarLocation.Y * DpiHelper.DpiScale))))
             {
                 result = true;
             }
 
             return result;
         }
 
-        private void SetPosition(uint x, uint y)
+        private void SetPosition(uint x = 0, uint y = 0)
         {
-            if (Settings.Instance.DesktopNavigationToolbarLocation != default &&
-                PointExistsOnScreen(Settings.Instance.DesktopNavigationToolbarLocation))
+            if (Settings.Instance.DesktopNavigationToolbarLocation != default)
             {
-                // use custom size
-                Top = Settings.Instance.DesktopNavigationToolbarLocation.Y / DpiHelper.DpiScale;
-                Left = Settings.Instance.DesktopNavigationToolbarLocation.X / DpiHelper.DpiScale;
+                Point desiredLocation = new Point()
+                {
+                    X = PctToAbsPos(Settings.Instance.DesktopNavigationToolbarLocation.X, ActualWidth, WindowManager.PrimaryMonitorSize.Width),
+                    Y = PctToAbsPos(Settings.Instance.DesktopNavigationToolbarLocation.Y, ActualHeight, WindowManager.PrimaryMonitorSize.Height)
+                };
+
+                if (PointExistsOnScreen(desiredLocation))
+                {
+                    Top = desiredLocation.Y;
+                    Left = desiredLocation.X;
+
+                    return;
+                }
             }
-            else
+
+            int sWidth;
+            int sHeight;
+            if (x != 0 && y != 0)
             {
                 // use size from wndproc and adjust for dpi
-                DpiHelper.TransformFromPixels(x, y, out int sWidth, out int sHeight);
-
-                SetPositionDefault(sWidth, sHeight);
+                DpiHelper.TransformFromPixels(x, y, out sWidth, out sHeight);
+            }
+            else
+            {
+                sWidth = WindowManager.PrimaryMonitorSize.Width;
+                sHeight = WindowManager.PrimaryMonitorSize.Height;
             }
+
+            SetPositionDefault(sWidth, sHeight);
         }
 
         private void SetPositionDefault(int screenWidth, int screenHeight)
         {
-            Top = screenHeight - Height - DEFAULT_Y;
-            Left = (screenWidth / 2) - (Width / 2);
+            Top = PctToAbsPos(DEFAULT_Y, ActualHeight, screenHeight);
+            Left = PctToAbsPos(DEFAULT_X, ActualWidth, screenWidth);
+        }
+
+        private double AbsToPctPos(double absolutePos, double size, double total)
+        {
+            // adjust to center point
+            return (absolutePos + size / 2) / total;
+        }
+
+        private double PctToAbsPos(double percentPos, double size, double total)
```

Comment:
It doesn't have to be in this PR but we should not use shortened or abbreviated terms unless it is universally known.
Abs and Pos are pretty universal, especially in development. Percent on the other hand has variations. At this point, with modern development and computers, abbreviations in code are more a limitation imposed by either habit or laziness. Rule of thumb, anything that you name (class, method or variable) should have a name who's length is somewhat related to its scope, lifetime and complexity. Shorter lifetime/scope ~= shorter, less descriptive name